### DISCUSSION TITLE
Add new View with Pos.Right() in event handler not working.

### DISCUSSION THREAD

**Discussion by Corn207** [tui-cs/Terminal.Gui#2472] [Q&A] [answered]
```
internal class MainWindow : Window
{
    public MainWindow()
    {
        var addBtn = new Button()
        {
            X = 0,
            Y = 0,
            Text = "Add"
        };

		var uselessBtn = new Button()
		{
			X = Pos.Right(addBtn),
			Y = 0,
			Text = "Useless"
		};

		addBtn.Clicked += () =>
        {
            var newBtn = new Button()
            {
                X = Pos.Right(uselessBtn) + 1,
                Y = 0,
                Text = "New"
            };
            Add(newBtn);
        };

        Add(addBtn, uselessBtn);
    }
}
```
![image](https://user-images.githubusercontent.com/63555670/228917544-0967a64d-9f17-4279-8936-2b320d66cfe4.png)
After pressed:
![image](https://user-images.githubusercontent.com/63555670/228917614-11e447da-df8b-4564-803e-a8ba889eb4a2.png)

**Comment by tznind** [comment, marked as answer]
Add a call to `LayoutSubviews ();` after adding the button:

```diff
var newBtn = new Button () {
	X = Pos.Right (uselessBtn) + 1,
	Y = 0,
	Text = "New"
};
Add (newBtn);
+LayoutSubviews ();
```

**Reply by Corn207** [reply]
Wow, thanks =))

**Comment by BDisp** [comment]
@tznind this can be avoided if the `Window.Redraw` call `LayoutSubviews ();`and then users doesn't needed to call it.
I ran the code in the `UICatalog` and when I press escape to exit I get this error in the v2_develop branch:

![imagem](https://user-images.githubusercontent.com/13117724/228927678-24d64834-d1d0-4749-b49b-cbb6eb4a6b14.png)

@tig something went wrong in your last merge.
This is the code I used:

```cs
		public override void Init ()
		{
			Application.Init ();
			Application.Run (new MainWindow ());
		}
		internal class MainWindow : Window {
			public MainWindow ()
			{
				var addBtn = new Button () {
					X = 0,
					Y = 0,
					Text = "Add"
				};

				var uselessBtn = new Button () {
					X = Pos.Right (addBtn),
					Y = 0,
					Text = "Useless"
				};

				addBtn.Clicked += (s, e) => {
					var newBtn = new Button () {
						X = Pos.Right (uselessBtn) + 1,
						Y = 0,
						Text = "New"
					};
					Add (newBtn);
					LayoutSubviews ();
				};

				Add (addBtn, uselessBtn);
			}
		}

```

**Reply by tig** [reply]
> @tznind this can be avoided if the `Window.Redraw` call `LayoutSubviews ();`and then users doesn't needed to call it. I ran the code in the `UICatalog` and when I press escape to exit I get this error in the v2_develop branch:
> 

I do not think `Redraw` should call `LayoutSubviews` - this is a cause performance issues (and spaghetti code) in v1.

In `v2` we will be enforcing that `Layout` is decoupled from `Drawing`. If, when `Redraw` is called layout is needed, we'll wait until the next `MainLoop` iteration to call `LayoutSubviews`.  

In `v2` is OK for the layout engine to call `Redraw`, but not the other way around.

**Reply by BDisp** [reply]
I think the `View.Redraw` is calling `LayoutSubviews` before draw his subviews and not calling `LayoutSubviews` to itself.
